build_runner_core/lib/src/generate/build_impl.dart

```diff
@@ -825,3 +828,13 @@ String _actionLoggerName(
       : primaryInput.uri;
   return '${phase.builderLabel} on $asset';
 }
+
+class BuildStepStageTracker implements StageTracker {
+  final BuilderActionTracker _tracker;
```

Comment:
We should just make this class implement StageTracker imo - that way we don't have to create a bunch of these wrapper objects